internal/pkg/describe/pipeline_status_test.go

```diff
@@ -0,0 +1,139 @@
+}
+
+var mockPipelineName = "pipeline-dinder-badgoose-repo"
+var mockParsedTime = func() *time.Time {
+	t, _ := time.Parse(time.RFC3339, "2020-02-02T15:04:05+00:00")
+	return &t
+}
+var mockPipelineState = &codepipeline.PipelineState{
+	PipelineName: mockPipelineName,
+	StageStates: []*codepipeline.StageState{
+		{
+			StageName:  "Source",
+			Status:     "Succeeded",
+			Transition: "ENABLED",
+		},
+		{
+			StageName:  "Build",
+			Status:     "Succeeded",
+			Transition: "ENABLED",
+		},
+		{
+			StageName:  "DeployTo-test",
+			Status:     "In Progress",
+			Transition: "ENABLED",
+		},
+		{
+			StageName:  "DeployTo-prod",
+			Status:     "Failed",
+			Transition: "DISABLED",
+		},
+	},
+	UpdatedAt: mockParsedTime(),
+}
+
+func TestPipelineStatusDescriber_Describe(t *testing.T) {
+	mockError := errors.New("some error")
+
+	testCases := map[string]struct {
+		setupMocks func(m pipelineStatusDescriberMocks)
+
+		expectedError  error
+		expectedOutput *PipelineStatus
+	}{
+		"wraps GetPipelineState error": {
+			setupMocks: func(m pipelineStatusDescriberMocks) {
+				m.pipelineStateGetter.EXPECT().GetPipelineState(pipelineName).Return(nil, mockError)
+			},
+			expectedError:  fmt.Errorf("get pipeline status: %w", mockError),
+			expectedOutput: nil,
+		},
+		"success": {
+			setupMocks: func(m pipelineStatusDescriberMocks) {
+				m.pipelineStateGetter.EXPECT().GetPipelineState(pipelineName).Return(mockPipelineState, nil)
+			},
+			expectedError:  nil,
+			expectedOutput: &PipelineStatus{*mockPipelineState},
+		},
+	}
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			mockPipelineStateGetter := mocks.NewMockpipelineStateGetter(ctrl)
+
+			mocks := pipelineStatusDescriberMocks{
+				pipelineStateGetter: mockPipelineStateGetter,
+			}
+			tc.setupMocks(mocks)
+
+			describer := &PipelineStatusDescriber{
+				pipelineName: pipelineName,
+				pipelineSvc:  mockPipelineStateGetter,
+			}
+
+			// WHEN
+			pipelineStatus, err := describer.Describe()
+
+			// THEN
+			if tc.expectedError != nil {
+				require.EqualError(t, err, tc.expectedError.Error())
+			} else {
+				require.Nil(t, err)
+				require.Equal(t, tc.expectedOutput, pipelineStatus, "expected output content match")
+			}
+		})
+	}
+}
+
+func TestPipelineStatusDescriber_String(t *testing.T) {
+	testCases := map[string]struct {
+		testPipelineStatus  *PipelineStatus
+		expectedHumanString string
+		expectedJSONString  string
+	}{
+		"correct output": {
+			testPipelineStatus: &PipelineStatus{*mockPipelineState},
+			expectedHumanString: `Pipeline Status
+
+  Stage             Status              Transition          
+  -----             ------              ----------
+  Source            Succeeded           ENABLED
+  Build             Succeeded           ENABLED
+  DeployTo-test     In Progress         ENABLED
+  DeployTo-prod     Failed              DISABLED
```

Comment:
Sweet, in a follow up PR can we possibly add colors to these statuses and transitions?
I'm thinking maybe: 
1. "Succeeded" => default
2. "In progress" => color.Emphasize
3. "Failed" => color.Red
4. "Enabled" => default
5. "Disabled" => color.Faint

Reply:
Yes!